### PR TITLE
Restructure RN to a single output file

### PR DESCRIPTION
**Pull request by mystiberry-okta**
## Description:
- All the new files for creating a single HTML page change log from the existing release notes.

### Partially Resolves:
 [OKTA-148503](https://oktainc.atlassian.net/browse/OKTA-148503)

### CODE REVIEW

File: _source/_posts/2017-04-17-angular-authentication-with-oidc.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ Angular:
 
 ## Run Your Angular Application
 
-The project is configured with [webpack dev server](https://webpack.github.io/docs/webpack-dev-server.html). To start it, make sure you're in the `ng-demo` directory, then run:
+The project is configured with [webpack dev server](https://webpack.js.org/). To start it, make sure you're in the `ng-demo` directory, then run:
```
Comment (mystiberry-okta) [inline _source/_posts/2017-04-17-angular-authentication-with-oidc.md]: @mraible This is the location that webpack's redirect gave me. Please let me know if you have a better one.

Comment (jmelberg-okta) [reply _source/_posts/2017-04-17-angular-authentication-with-oidc.md]: @mystiberry-okta: This one might make more sense for developers -> [webpack-dev-server](https://github.com/webpack/webpack-dev-server)

**Comment by jmelberg-okta** [comment]
**Question: Does each individual file need title: metadata at top of file? excerpt: ?**
Not necessarily. But I'd suggest validating each file to contains a `title` attribute. This will allow us to modify the `index.md` template to use the `title` attribute, instead of assuming it is contained within the file.

For example, we can use this template syntax:
```
<div>
{% assign sorted = site.change-log | sort: 'date' | reverse %}
{% for file in sorted %}
  {% if file.id contains "/index" %}{% continue %}{% endif %}
  <b>{{file.title}}</b>
  <i>{{file.excerpt}}</i>
  {{ file.content }}
{% endfor %}
</div>
```

File: _source/_posts/2017-04-17-angular-authentication-with-oidc.md (hunk 1 above)
Comment (mystiberry-okta) [reply _source/_posts/2017-04-17-angular-authentication-with-oidc.md]: @mraible https://github.com/webpack/webpack-dev-server okay?

**Comment by mystiberry-okta** [comment]
Great suggestions! For now, leaving title & excerpt in each individual file, JIRA for these (and probably more) improvements: https://oktainc.atlassian.net/browse/OKTA-154009

File: _config.yml [hunk 2]
```diff
@@ -76,6 +76,9 @@ collections:
     output: true
   quickstarts:
     output: true
+  change-log:
+    output: true
```
Comment (bdemers) [inline _config.yml]: This should likely be false, as we don't want the individual files rendered (we only want them present in the aggregated file)

Comment (mystiberry-okta) [reply _config.yml]: Ack, thanks!

**Comment by mystiberry-okta** [comment]
@bdemers That command failed for me but I read https://github.com/jekyll/jekyll-redirect-from and should be able to implement. Thanks!

**Comment by bdemers** [comment]
@mystiberry-okta that command would have output something like this (which should save you the trouble of figuring out which redirects need to be added) :

```
  - "/docs/platform-release-notes/platform-release-notes2016-23.html"
  - "/docs/platform-release-notes/platform-release-notes2016-24.html"
  - "/docs/platform-release-notes/platform-release-notes2016-25.html"
  - "/docs/platform-release-notes/platform-release-notes2016-26.html"
  - "/docs/platform-release-notes/platform-release-notes2016-27.html"
  - "/docs/platform-release-notes/platform-release-notes2016-28.html"
  - "/docs/platform-release-notes/platform-release-notes2016-29.html"
  - "/docs/platform-release-notes/platform-release-notes2016-30.html"
  - "/docs/platform-release-notes/platform-release-notes2016-31.html"
  - "/docs/platform-release-notes/platform-release-notes2016-33.html"
  - "/docs/platform-release-notes/platform-release-notes2016-34.html"
  - "/docs/platform-release-notes/platform-release-notes2016-35.html"
  - "/docs/platform-release-notes/platform-release-notes2016-36.html"
  - "/docs/platform-release-notes/platform-release-notes2016-37.html"
  - "/docs/platform-release-notes/platform-release-notes2016-39.html"
  - "/docs/platform-release-notes/platform-release-notes2016-40.html"
  - "/docs/platform-release-notes/platform-release-notes2016-41.html"
  - "/docs/platform-release-notes/platform-release-notes2016-43.html"
  - "/docs/platform-release-notes/platform-release-notes2016-45.html"
  - "/docs/platform-release-notes/platform-release-notes2016-46.html"
  - "/docs/platform-release-notes/platform-release-notes2016-47.html"
  - "/docs/platform-release-notes/platform-release-notes2016-49.html"
  - "/docs/platform-release-notes/platform-release-notes2016-50.html"
  - "/docs/platform-release-notes/platform-release-notes2016-51.html"
  - "/docs/platform-release-notes/platform-release-notes2016-52.html"
  - "/docs/platform-release-notes/platform-release-notes2016-index.html"
  - "/docs/platform-release-notes/platform-release-notes2017-02.html"
  - "/docs/platform-release-notes/platform-release-notes2017-03.html"
  - "/docs/platform-release-notes/platform-release-notes2017-04.html"
  - "/docs/platform-release-notes/platform-release-notes2017-05.html"
  - "/docs/platform-release-notes/platform-release-notes2017-09.html"
  - "/docs/platform-release-notes/platform-release-notes2017-10.html"
  - "/docs/platform-release-notes/platform-release-notes2017-11.html"
  - "/docs/platform-release-notes/platform-release-notes2017-12.html"
  - "/docs/platform-release-notes/platform-release-notes2017-15.html"
  - "/docs/platform-release-notes/platform-release-notes2017-16.html"
  - "/docs/platform-release-notes/platform-release-notes2017-17.html"
  - "/docs/platform-release-notes/platform-release-notes2017-18.html"
  - "/docs/platform-release-notes/platform-release-notes2017-19.html"
  - "/docs/platform-release-notes/platform-release-notes2017-20.html"
  - "/docs/platform-release-notes/platform-release-notes2017-21.html"
  - "/docs/platform-release-notes/platform-release-notes2017-22.html"
  - "/docs/platform-release-notes/platform-release-notes2017-23.html"
  - "/docs/platform-release-notes/platform-release-notes2017-24.html"
  - "/docs/platform-release-notes/platform-release-notes2017-25.html"
  - "/docs/platform-release-notes/platform-release-notes2017-26.html"
  - "/docs/platform-release-notes/platform-release-notes2017-27.html"
  - "/docs/platform-release-notes/platform-release-notes2017-28.html"
  - "/docs/platform-release-notes/platform-release-notes2017-29.html"
  - "/docs/platform-release-notes/platform-release-notes2017-30.html"
  - "/docs/platform-release-notes/platform-release-notes2017-31.html"
  - "/docs/platform-release-notes/platform-release-notes2017-32.html"
  - "/docs/platform-release-notes/platform-release-notes2017-33.html"
  - "/docs/platform-release-notes/platform-release-notes2017-34.html"
  - "/docs/platform-release-notes/platform-release-notes2017-35.html"
  - "/docs/platform-release-notes/platform-release-notes2017-36.html"
  - "/docs/platform-release-notes/platform-release-notes2017-38.html"
  - "/docs/platform-release-notes/platform-release-notes2017-40.html"
  - "/docs/platform-release-notes/platform-release-notes2017-41.html"
  - "/docs/platform-release-notes/platform-release-notes2017-42.html"
  - "/docs/platform-release-notes/platform-release-notes2017-43.html"
  - "/docs/platform-release-notes/platform-release-notes2017-44.html"
  - "/docs/platform-release-notes/platform-release-notes2017-45.html"
  - "/docs/platform-release-notes/platform-release-notes2017-46.html"
  - "/docs/platform-release-notes/platform-release-notes2017-47.html"
  - "/docs/platform-release-notes/platform-release-notes2017-49.html"
  - "/docs/platform-release-notes/platform-release-notes2017-50.html"
  - "/docs/platform-release-notes/platform-release-notes2017-index.html"
```

Granted, I only searched for files in the dist folder that have `platform-release-notes20` in the path.

File: _config.yml (hunk 2 above)
Comment (mystiberry-okta) [reply _config.yml]: @bdemers So changing that value to false makes the aggregated file not render...?

Comment (bdemers) [reply _config.yml]: no, it makes the individual files of the collection not render

**Comment by mystiberry-okta** [comment]
Oh! Yeah, all the files in that directory need a redirect. No worries. Very glad it can just be done in the aggregate file, thanks!

File: _source/_posts/2017-04-17-angular-authentication-with-oidc.md (hunk 1 above)
Comment (jmelberg-okta) [reply _source/_posts/2017-04-17-angular-authentication-with-oidc.md]: ^ This is being picked up in another PR, as it is a merge blocker. Please make sure to rebase.

File: _config.yml (hunk 2 above)
Comment (mystiberry-okta) [reply _config.yml]: So I tested it four times, restarting local build after saving each time. if `true`, I can access the index.html file. If `false`, I cannot. http://localhost:4000/docs/change-log/index.html or http://localhost:4000/docs/change-log/

Comment (bdemers) [reply _config.yml]: Oh, easy fix :)  your new index page cannot be in the same directory as the collection ;)

Comment (mystiberry-okta) [reply _config.yml]: Ack, thanks!

**Comment by mystiberry-okta** [comment]
A few more tweaks, not quite ready for review yet.

File: _source/_posts/2017-04-17-angular-authentication-with-oidc.md (hunk 1 above)
Comment (mystiberry-okta) [reply _source/_posts/2017-04-17-angular-authentication-with-oidc.md]: I reverted so this won't even be in this pr. Thanks!

**Comment by mystiberry-okta** [comment]
Status: Can't build index.html with output: false, but if output: true, all the individual files build. Demers to take a look and see what's wrong.

File: _config.yml (hunk 2 above)
Comment (jmelberg-okta) [reply _config.yml]: Leaving the output to true is required only for the `change-log` collection. The `change-log` `index.md` file loops over a new collection: `release-notes` and renders a single file.

**Comment by jmelberg-okta** [comment]
@bdemers & @mystiberry-okta:

We now have two collections: `release-notes` and `change-log`. When adding a new file during the **weekly release**, place it into the `_release-notes` dir. These files are **not** individually rendered.

Change log: -> `https://developer.okta.com/docs/change-log/`

File: _source/_reference/api_reference.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ---
 layout: docs_page
 weight: 3
-title: API Reference
-link: /docs/api/resources/apps.html
-excerpt: Documentation for all available Okta REST API endpoints. Includes sample requests and responses.
+title: Management API
```
Comment (jmelberg-okta) [inline _source/_reference/api_reference.md]: Curious why is this included with this PR? I believe this change breaks our tests.

File: _source/_docs/platform-release-notes/platform-release-notes2016-23.md [hunk 4]
```diff
@@ -1,22 +1,8 @@
 ---
 layout: docs_page
+redirect_to:
```
Comment (jmelberg-okta) [inline _source/_docs/platform-release-notes/platform-release-notes2016-23.md]: Instead, lets put this path inside of the `index.md` file and use `redirect_from`:

> redirect_from:
>   - /docs/platform-release-notes/platform-release-notes2016-23

File: _source/_includes/footer.html [hunk 5]
```diff
@@ -28,7 +28,7 @@ <h4>More Info</h4>
                 <ul class="Footer-links">
                     <li><a target="_blank" href="{{ site.external_domain }}/integrate-with-okta/">Integrate with Okta</a></li>
                     <li><a href="/blog/">Blog</a></li>
-                    <li><a href="/docs/platform-release-notes/platform-release-notes.html">Release Notes</a></li>
+                    <li><a href="/docs/change-log/index.html">Change Log</a></li>
```
Comment (jmelberg-okta) [inline _source/_includes/footer.html]: Can we use the pretty URL here instead? `/docs/change-log/`

File: _source/_docs/api/getting_started/releases-at-okta.md [hunk 6]
```diff
@@ -18,7 +18,7 @@ These updates are non-breaking because they are additive. Follow [the compatibil
 when additive changes are made. 
 Breaking changes such as removing or renaming an attribute are released as a new version of the API, and Okta provides a migration path for new API versions.
 
-Changes, regardless of lifecycle stage, are always reported in the [Okta API Release Notes Release Notes](/docs/platform-release-notes/platform-release-notes.html).
+Changes, regardless of lifecycle stage, are always reported in the [Okta API Products Change Log](/docs/change-log/index.html).
```
Comment (jmelberg-okta) [inline _source/_docs/api/getting_started/releases-at-okta.md]: Lets remove the `index.html` file here to use the pretty URL: `/docs/change-log/`

File: _source/_docs/platform-release-notes/platform-release-notes2016-23.md (hunk 4 above)
Comment (jmelberg-okta) [reply _source/_docs/platform-release-notes/platform-release-notes2016-23.md]: We can actually remove this, since we already have a `redirect_from` in our `index.md` page.

File: _source/_reference/api_reference.md (hunk 3 above)
Comment (jmelberg-okta) [reply _source/_reference/api_reference.md]: Per offline convo, we've updated the tests to account for this change.

File: _source/_reference/z-change-log.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+---
+layout: docs_page
+weight: 6
+title: Change Log
+link: /docs/change-log/index.html
+excerpt: Review the changes to Okta API Products by release.
```
Comment (jmelberg-okta) [inline _source/_reference/z-change-log.md]: Where is this file used?

File: _source/_docs/api/getting_started/releases-at-okta.md (hunk 6 above)
Comment (mystiberry-okta) [reply _source/_docs/api/getting_started/releases-at-okta.md]: ack.

**Comment by bdemers** [comment]
We still need to delete the old release note files

**Comment by jmelberg-okta** [comment]
@bdemers I'm picking this up on a separate PR.

**Comment by bdemers** [comment]
The Authentication Reference/Authentication API changes should be moved to a different PR

File: _source/_reference/z-change-log.md (hunk 7 above)
Comment (mystiberry-okta) [reply _source/_reference/z-change-log.md]: Ah, this is where the content for the landing page comes from! That excerpt is printed under the link.

**Comment by bdemers** [comment]
@jmelberg-okta I'm -1 for adding them in a separate PR, they should be added to this one so they can be merged at the same time. it should be trivial to add to this commit

**Comment by mystiberry-okta** [comment]
@bdemers we're separating because of the localhost:4000/developer.okta.com problem. @jmelberg-okta can confirm whether we really need to (we've changed course a few times on the redirect).

**Comment by mystiberry-okta** [comment]
@bdemers I'd rather leave the label changes in. It caught my attention because of changes I had to make for the RN > Change Log.

File: _source/_includes/footer.html (hunk 5 above)
Comment (mystiberry-okta) [reply _source/_includes/footer.html]: Ack

**Comment by bdemers** [comment]
I still think we should pull them out of this PR, and keep this one focused, there is already a lot going on with this one. and if we have to back it out, we don't want to back out those unrelated changes.

File: _source/_docs/platform-release-notes/platform-release-notes2016-23.md [hunk 8]
```diff
@@ -1,8 +1,22 @@
...
 layout: docs_page
-redirect_to:
-- /docs/change-log/
 title: Platform Release Notes June 8, 2016
 ---
-You should be redirected to the Change Log in a moment.
-If not, [click here](/docs/change-log/).
\ No newline at end of file
+
+# Release 2016.23
```
Comment (jmelberg-okta) [inline _source/_docs/platform-release-notes/platform-release-notes2016-23.md]: @mystiberry-okta Why did we revert this?

**Comment by mystiberry-okta** [comment]
@bdemers I'd rather not, for me it's part of the RN > Change Log change and there's no technical reason to pull it out. I want the changes to go out together, anyway.

**Comment by jmelberg-okta** [comment]
@bdemers & @mystiberry-okta:

Lets keep those changes in to avoid the headache of splitting it from a commit.

@mystiberry-okta Can you please update the description of this PR to include the title changes? That way we can trace back and find when/where the changes were made.
